### PR TITLE
Skip any elements in the range that are already second-order

### PR DESCRIPTION
**Pull request by jwpeterson**
>  however I don't think this will be as simple as just deleting the error message

OK, so I did not delete the error message exactly, it's still there for Elems that are neither FIRST nor SECOND-order, but simply skipping elements that are already SECOND order worked for my test case. I'll see about creating a unit test for this, but in the meantime wanted to get the proposed fix out there.

Refs #3533

### CODE REVIEW

File: src/mesh/unstructured_mesh.C [hunk 1]
```diff
@@ -1412,7 +1412,16 @@ UnstructuredMesh::all_second_order_range (const SimpleRange<element_iterator> &
...
   for (auto & lo_elem : range)
     {
-      // make sure it is linear order
+      // Skip elements in the range that are already SECOND-order.
+      // Ordinarily one does not have a mixture of FIRST and
+      // SECOND-order elements in an all-2D or all-3D mesh, since
+      // FIRST and SECOND-order elements can't be (conforming)
+      // neighbors, but in a mixed-dimension element mesh, this is
+      // possible.
+      if (lo_elem->default_order() == SECOND)
```
Comment (roystgnr) [inline src/mesh/unstructured_mesh.C:1421]: Should this be `!= FIRST`, since we may have higher-order?

**Review by roystgnr — CHANGES_REQUESTED** [review]
That's the only issue I see.

File: src/mesh/unstructured_mesh.C (hunk 1 above)
Comment (jwpeterson) [reply src/mesh/unstructured_mesh.C:1421]: The `!= FIRST` case is still handled by the subsequent  `libmesh_error_msg()`. In other words, we only want to `continue` for `SECOND`-order elements, `THIRD`, `FOURTH`, etc. we'll still throw an error.

**Review by roystgnr — APPROVED** [review]
Makes sense; I guess the average user here might be surprised to see a *down* conversion or might be surprised to see *no* conversion; there's not one obvious semantics for a third-order element.

**Comment by roystgnr** [comment]
This should be enough to close #3533?

**Pull request merged**